tools/setup_helpers/ib_detect.py

```diff
@@ -0,0 +1,76 @@
+import os
+import subprocess
+import re
+
+
+WITH_IB_DEVICES = False
+IB_DEVINFO_CMD = "ibv_devinfo"
+
+
+def get_command_path(command):
+    """
+    Helper function that get the full path of a given linux command
+    """
+    def excutable(command_path):
+        return os.path.isfile(command_path) and os.access(command_path, os.X_OK)
+
+    for path in os.environ["PATH"].split(os.pathsep):
+        command_path = os.path.join(path, command)
+        if excutable(command_path):
+            return command_path
+
+    return None
+
+
+def detect_ib_devices():
+    """
+    Helper function that detects if there are Infiniband devices on the host,
+    and returns the number of IB devices detected or None for failure to detect
+    """
+    try:
+        full_cmd_path = get_command_path(IB_DEVINFO_CMD)
+        if not full_cmd_path:
+            return None
+        out = subprocess.check_output([full_cmd_path, "--list"])
+        # find the first line of the output
+        # The outpyt should be either:
+        #
+        # > ibv_devinfo --list
+        # 0 HCAs founds:
+        #
+        # or
+        #
+        # > ibv_devinfo --list
+        # 4 HCAs found:
+        #   mlx5_3
+        #   mlx5_2
+        #   mlx5_1
+        #   mlx5_0
+        first_line = out.decode().split('\n')[0]
+        res = re.findall("\d+", first_line)
+        if len(res) != 1:
+            raise Exception("-- IB_detect: unexpected parsing error while "
+                            "trying to find the number of available devices.")
+        return int(res[0])
```

Comment:
Can we just try to find and run the executable without any parsing? Having no devices doesn’t mean that you don’t want IB - maybe you’re building this in a cluster environment where your nose only has the libraries and the cluster workers have IB.

Reply:
That's fine too